### ISSUE TITLE
Loss Function

### ISSUE THREAD

**Issue by haoming-codes** [cvignac/DiGress#5]
The loss function used in the training is `CrossEntropy(model(G_t), G)`, whereas traditionally in diffusion models it is `CrossEntropy(model(G_t), G_{t-1})`. Does using the latter make the training worse in your experience? Thank you!

**Comment by cvignac** [comment]
It's important to use `model(G_t), G` and not `model(G_t), G_{t-1}`, but what we are doing is what most model do in fact. Check section 2 of the paper, the reason for this is explained.

Best,
Clément

**Comment by haoming-codes** [comment]
Hi Clément, Thanks for the pointer. I'm afraid I still could use your help after reading Sec. 2. To my understanding:

- Ho (2020)'s loss function is `L_simple`
- Nichol and Dhariwal (2021)'s loss function is  `L_simple + L_vb`
- Austin (2021)'s loss function is `L_vb + CrossEntropy(model(G_t), G)`
- Your loss function is `CrossEntropy(model(G_t), G)`

It seems to me that the idea that the model should be encouraged to "do as much as is can" is a novelty of Austin (2021), whereas `CrossEntropy(model(G_t), G_{t-1})` is more similar to what a diffusion model is designed for.  Can you point me to other diffusion papers where only loss of the form`CrossEntropy(model(G_t), G)` is used? Many thanks!

**Comment by cvignac** [comment]
If you look at `L_simple` for example, it tries to predict the loss that has been added to the clean sample to create z_t. 

Up to a substraction, predicting the loss is the same as predicting the clean sample, so Ho et al. 2020 predict the clean sample, not z^{t-1}.

It's similar for the other ones.

**Comment by najwalb** [comment]
I have a related question on the derivation of the loss (let me know if I should make a new issue).

Why do you train on `CrossEntropy(model(G_t), G)` as opposed to `L_vb` (general loss) or `L_vb + CrossEntropy(model(G_t), G)` (like in Austin (2021))?

Ho (2020) trains on `L_simple` but in the case of Gaussian noise `L_simple`, which is the MSE between the predicted and real noised mean (eq 8) or the denoising score matching objective (eq 12),  **is essentially an unweighted `L_vb`**. This simplification comes from the KL divergence of two Gaussians which leads to Eq 8.

Is the cross-entropy score you're using also an unweighted `L_vb`? Austin (2021) discuss this a little bit in section 3.4., paragraph 2, but I didn't understand whether the answer is yes/no. 

Thanks in advance! Great implementation btw, following through the code was easy:)

**Comment by cvignac** [comment]
Hello Najwalb, 
we trained using CrossEntropy loss for simplicity, and as a way to highlight that denoising diffusion models reduce generation tasks to regression or classification tasks. 

It's not so clear how it relates to `L_{vb}`. As Austin discuss in 3.4, we know that the optimal solutions of the two loss functions are the same, but it is not clear that for arbitrary `q`, `L_simple` can be written `\sum_t \gamma_t L_vb^t`. It would be interesting to explore this further.

**Comment by haoming-codes** [comment]
Thanks @cvignac. I'm curious to know if you are aware other diffusion papers in which the role of the model is to transform the data, as opposed to to predict the score (in score-based models) or the noise (in DDPM).

**Comment by cvignac** [comment]
Hello,
There is the Austin paper of course, and then there are several papers such as https://arxiv.org/pdf/2210.02303.pdf
which use v-parametrization, which is a mix between predicting the noise and the clean data.

I don't think that these things are fundamentally different. Since clean_data = noisy_data - noise, predicting the noise and the clean data is pretty much the same. Then it's a matter of stability of the inverse diffusion steps. 

For discrete diffusion, predicting the clean data makes sense because it is simply one class, while the noise is the difference of two classes. That's why we chose this parametrization.

Best,
Clement
